### PR TITLE
feat: show more details when no test files found

### PR DESCRIPTION
**Pull request by togami2864**
#1032 
Enable vitest to show 

- include
- watchIgnore

when no test files found.
screen shot

### CODE REVIEW

File: packages/vitest/src/node/core.ts [hunk 1]
```diff
@@ -114,12 +114,17 @@ export class Vitest {
     )
 
     if (!files.length) {
-      if (this.config.passWithNoTests)
-        this.log('No test files found\n')
+      const exitCode = this.config.passWithNoTests ? 0 : 1
+      if (this.config.passWithNoTests) { this.log(`No test files found, existing code with ${exitCode}\n`) }
```
Comment (dammy001) [inline packages/vitest/src/node/core.ts:118]:
```suggestion
      if (this.config.passWithNoTests) { this.log(`No test files found, exiting code with ${exitCode}\n`) }
```

File: docs/config/index.md [hunk 2]
```diff
@@ -318,6 +318,27 @@ Isolate environment for each test file
 
 Coverage options
 
+### testNamePattern
+
+- **Type** `string | RegExp`
+
+Run tests with full names matching the pattern.
+If you add `do not run` to this property, tests containing the word `do not run` in the test name will be skipped.
```
Comment (sheremet-va) [inline docs/config/index.md:326]: I feel like the opposite will happen? Only tests containing "do not run" will run, no?

File: packages/vitest/src/utils/base.ts [hunk 3]
```diff
@@ -112,3 +112,11 @@ export function assertTypes(value: unknown, name: string, types: string[]): void
   const pass = types.includes(receivedType)
   if (!pass) throw new TypeError(`${name} value must be ${types.join(' or ')}, received "${receivedType}"`)
 }
+
+/**
+ * joinArrayItems
+ * convert Array<T> to a single string to display in stdout
+ */
+export function joinArrayItems(items: unknown[]): string {
```
Comment (sheremet-va) [inline packages/vitest/src/utils/base.ts:120]: just out of interest, what is the difference between just calling `.join`? I see all test cases will return the same if called with `.join`

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/utils/base.ts (hunk 3 above)
Comment (togami2864) [reply packages/vitest/src/utils/base.ts]: Fixed to use `join` since `include` and `watchIgnore`  are absolutely the array. Thanks.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**